lib/web_ui/lib/src/engine/text_editing/input_action.dart

```diff
@@ -0,0 +1,155 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import '../browser_detection.dart';
+import '../dom.dart';
+
+/// Various input action types used in text fields.
+///
+/// These types are coming from Flutter's [TextInputAction]. Currently, the web doesn't
+/// support all the types. We fallback to [EngineInputAction.none] when Flutter
+/// sends a type that isn't supported.
+abstract class EngineInputAction {
+  const EngineInputAction();
+
+  static EngineInputAction fromName(String name) {
+    switch (name) {
+      case 'TextInputAction.continueAction':
+      case 'TextInputAction.next':
+        return next;
+      case 'TextInputAction.previous':
+        return previous;
+      case 'TextInputAction.done':
+        return done;
+      case 'TextInputAction.go':
+        return go;
+      case 'TextInputAction.newline':
+        return enter;
+      case 'TextInputAction.search':
+        return search;
+      case 'TextInputAction.send':
+        return send;
+      case 'TextInputAction.emergencyCall':
+      case 'TextInputAction.join':
+      case 'TextInputAction.none':
+      case 'TextInputAction.route':
+      case 'TextInputAction.unspecified':
+      default:
+        return none;
+    }
+  }
+
+  /// No input action
+  static const NoInputAction none = NoInputAction();
+
+  /// Action to go to next
+  static const NextInputAction next = NextInputAction();
+
+  /// Action to go to previous
+  static const PreviousInputAction previous = PreviousInputAction();
+
+  /// Action to be finished
+  static const DoneInputAction done = DoneInputAction();
+
+  /// Action to Go
+  static const GoInputAction go = GoInputAction();
+
+  /// Action to insert newline
+  static const EnterInputAction enter = EnterInputAction();
+
+  /// Action to search
+  static const SearchInputAction search = SearchInputAction();
+
+  /// Action to send
+  static const SendInputAction send = SendInputAction();
+
+
+  /// The HTML `enterkeyhint` attribute to be set on the DOM element.
+  ///
+  /// This HTML attribute helps the browser decide what kind of keyboard action
+  /// to use for this text field
+  ///
+  /// For various `enterkeyhint` values supported by browsers, see:
+  /// <https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/enterkeyhint>.
+  String? get enterkeyhintAttribute;
+
+  /// Given a [domElement], set attributes that are specific to this input action.
+  void configureInputAction(DomHTMLElement domElement) {
+    if (enterkeyhintAttribute == null) {
+      return;
+    }
+
+    // Only apply `enterkeyhint` in mobile browsers so that the right virtual
+    // keyboard shows up.
```

Comment:
Would it be harmful to set it on desktop browsers? Some Windows and ChromeOS laptops these days are convertible to tablet mode with on-screen keyboards.

Reply:
Probably not, though I was just following the pattern from the input_type.dart file, which does this check. I'll happily change this if you want, but would it be best to change it in both places?

Reply:
Ah, if this is an existing pattern, it's OK to follow it for now. At some point we should revise our strategy for touch-enabled Window and ChromeOS devices.